### ISSUE TITLE
CentOS 8.1 dkms also fails

### ISSUE THREAD

**Issue by wyoung** [zfsonlinux/zfs#9866]
This appears to be distinct from #9856. That's about the kmod version, this is about the default dkms version.

### System information

Type | Version/Name
 --- | --- 
Distribution Name	|CentOS
Distribution Version	| 8.1
Linux Kernel	| 4.18.0-147.3.1.el8_1.x86_64
Architecture	| x86_64
ZFS Version	| 0.8.x
SPL Version	| n/a

### Describe how to reproduce the problem

Installation of the repo gives partial success:

```
# yum -y install http://download.zfsonlinux.org/epel/zfs-release.el8_1.noarch.rpm

[snip]

  Running scriptlet: zfs-release-1-8.1.noarch                               1/1 
error: can't create transaction lock on /var/lib/rpm/.rpm.lock (Resource temporarily unavailable)
error: /etc/pki/rpm-gpg/RPM-GPG-KEY-zfsonlinux: key 1 import failed.
warning: %post(zfs-release-1-8.1.noarch) scriptlet failed, exit status 1

Error in POSTIN scriptlet in rpm package zfs-release
  Verifying        : zfs-release-1-8.1.noarch                               1/1 

Installed:
  zfs-release-1-8.1.noarch   
```

You may be able to avoid the error above by moving the GPG key instalation into the `%posttrans` script. The `%post` script happens [surprisingly early](https://docs.fedoraproject.org/en-US/packaging-guidelines/Scriptlets/#ordering) in the yum/dnf instalation process!

Then on attempting to do anything with the repo:

```
# yum upgrade
ZFS on Linux for EL8 - dkms                     1.1 kB/s | 306  B     00:00    
Failed to download metadata for repo 'zfs'
Error: Failed to download metadata for repo 'zfs'
```

This breaks not only the ZFS repo but also effective use of the regular CentOS ones, since the zfs repo gets tried, fails, and breaks every yum/dnf transaction thereafter:

```
# yum install perl
CentOS-8 - AppStream                            6.9 kB/s | 4.3 kB     00:00    
CentOS-8 - Base                                  11 kB/s | 3.8 kB     00:00    
CentOS-8 - Extras                               5.4 kB/s | 1.5 kB     00:00    
Extra Packages for Enterprise Linux Modular 8 -  46 kB/s |  19 kB     00:00    
Extra Packages for Enterprise Linux 8 - x86_64   58 kB/s |  19 kB     00:00    
ZFS on Linux for EL8 - dkms                     1.0 kB/s | 306  B     00:00    
Failed to download metadata for repo 'zfs'
Error: Failed to download metadata for repo 'zfs'
```

Removing the `zfs-release` package fixes the symptom.

**Comment by wyoung** [comment]
We've tried this on another machine, and the problem appears to have gone away.

**Comment by aaronbaugher** [comment]
I still get that same error on a fresh 8.1 install:

```
Running scriptlet: zfs-release-1-8.1.noarch                                     1/1 
error: can't create transaction lock on /var/lib/rpm/.rpm.lock (Resource temporarily unavailable)
error: /etc/pki/rpm-gpg/RPM-GPG-KEY-zfsonlinux: key 1 import failed.
warning: %post(zfs-release-1-8.1.noarch) scriptlet failed, exit status 1

Error in POSTIN scriptlet in rpm package zfs-release
  Verifying        : zfs-release-1-8.1.noarch                                     1/1 

Installed:
  zfs-release-1-8.1.noarch                                                            
```

**Comment by tonyhutter** [comment]
@aaronbaugher despite the error, did /etc/yum.repod.d/zfs.repo get installed?

**Comment by aaronbaugher** [comment]
@tonyhutter Yes, that's there. If I go on to the next step, it seems like the key is there, but only partially:

'''
# gpg --quiet --with-fingerprint /etc/pki/rpm-gpg/RPM-GPG-KEY-zfsonlinux
pub   rsa2048 2013-03-21 [SC]
uid           ZFS on Linux <zfs@zfsonlinux.org>
sub   rsa2048 2013-03-21 [E]
'''

I went ahead and followed the kAPI instructions, editing zfs.repo and doing `yum install zfs`, and that went fine, and I'm able to `modprobe zfs` and run `zpool`. So the only problem may be with the signing key.

**Comment by tangentsoft** [comment]
@aaronbaugher: To be clear, this is really two separate bugs, the GPG key installation complaint and then the breakage of yum/dnf in general once the repo is installed. I closed it when the second problem disappeared for us, since that's the one that effectively prevented us from using ZFS. 

The GPG error isn't as serious because it can be easily fixed the next time you try to use the repo manually: it asks if it can install the key and you say "yes."

Still, the `%posttrans` solution I suggested above should still be done, also.
